segmentstore/storage/src/test/java/io/pravega/segmentstore/storage/chunklayer/SystemJournalOperationsTests.java

```diff
@@ -744,6 +743,226 @@ void testScenarioWithFlakySnapshotInfoStore(TestMethod test, TestScenarioProvide
+        TestUtils.checkSegmentLayout(instance2.metadataStore, testSegmentName, new long[] { 10});
+        TestUtils.checkChunksExistInStorage(testContext.chunkStorage, instance2.metadataStore, testSegmentName);
+        val segmentMetadata2 = TestUtils.getSegmentMetadata(instance2.metadataStore, testSegmentName);
+        Assert.assertEquals("A", segmentMetadata2.getFirstChunk());
+        Assert.assertEquals("A", segmentMetadata2.getLastChunk());
+        Assert.assertEquals(0, segmentMetadata2.getFirstChunkStartOffset());
+        Assert.assertEquals(0, segmentMetadata2.getLastChunkStartOffset());
+
+        // Bootstrap a new instance.
+        @Cleanup
+        val instance3 =  new TestInstance(testContext, 3);
+        instance3.bootstrap();
+        instance3.validate();
+        TestUtils.checkSegmentBounds(instance3.metadataStore, testSegmentName, 2, 10);
+        TestUtils.checkSegmentLayout(instance3.metadataStore, testSegmentName, new long[] { 10});
+        TestUtils.checkChunksExistInStorage(testContext.chunkStorage, instance3.metadataStore, testSegmentName);
+        val segmentMetadata3 = TestUtils.getSegmentMetadata(instance3.metadataStore, testSegmentName);
+        Assert.assertEquals("A", segmentMetadata3.getFirstChunk());
+        Assert.assertEquals("A", segmentMetadata3.getLastChunk());
+        Assert.assertEquals(0, segmentMetadata3.getFirstChunkStartOffset());
+        Assert.assertEquals(0, segmentMetadata3.getLastChunkStartOffset());
+        instance3.truncate(testSegmentName, 3);
+
+        // Zombie Truncate
+        instance2.writeZombieRecord(SystemJournal.TruncationRecord.builder()
+                .offset(4)
+                .startOffset(0)
+                .segmentName(testSegmentName)
+                .firstChunkName("A")
+                .build());
+
+        // Bootstrap a new instance.
+        @Cleanup
+        val instance4 =  new TestInstance(testContext, 4);
+        instance4.bootstrap();
+        TestUtils.checkSegmentBounds(instance4.metadataStore, testSegmentName, 3, 10);
+        TestUtils.checkSegmentLayout(instance4.metadataStore, testSegmentName, new long[] { 10});
+        TestUtils.checkChunksExistInStorage(testContext.chunkStorage, instance4.metadataStore, testSegmentName);
+        val segmentMetadata4 = TestUtils.getSegmentMetadata(instance4.metadataStore, testSegmentName);
+        Assert.assertEquals("A", segmentMetadata4.getFirstChunk());
+        Assert.assertEquals("A", segmentMetadata4.getLastChunk());
+        Assert.assertEquals(0, segmentMetadata4.getFirstChunkStartOffset());
+        Assert.assertEquals(0, segmentMetadata4.getLastChunkStartOffset());
+    }
+
+    /**
+     * Test zombie scenario with multiple chunks.
+     * @throws Exception Exception if any.
+     */
+    @Test
+    public void testZombieScenarioMultipleChunks() throws Exception {
+        val testContext = new TestContext(CONTAINER_ID);
```

Comment:
`@Cleanup` for all these `TestContext` instances. They have a `close` method so they should be properly cleaned up.